controller/engine_controller.go

```diff
@@ -1230,8 +1242,12 @@ func handleRestoreError(log logrus.FieldLogger, engine *longhorn.Engine, rsMap m
 
 	for _, re := range taskErr.ReplicaErrors {
 		if status, exists := rsMap[re.Address]; exists {
-			if strings.Contains(re.Error(), "already in progress") || strings.Contains(re.Error(), "already restored backup") {
+			if strings.Contains(re.Error(), "already in progress") || strings.Contains(re.Error(), "already restored backup") || strings.Contains(re.Error(), restoreGetLockFailedMsg) {
 				log.WithError(re).Debugf("Ignore restore error from replica %v", re.Address)
+				if strings.Contains(re.Error(), restoreGetLockFailedMsg) {
```

Comment:
Suggest to have a separate if statement instead of wrapping in the `if strings.Contains(re.Error(), "already in progress") || strings.Contains(re.Error(), "already restored backup") || strings.Contains(re.Error(), restoreGetLockFailedMsg)`.

Reply:
Modified. More clear. Thanks.